### PR TITLE
Convert comments to php docs in some cases

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Convert comments to php docs in some cases
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11747)

### CODE REVIEW

**Comment by MathiasReker** [comment]
I would like to work on the docs, so it could be nice to have this merged asap if it is OK :) @PierreRambaud  @jolelievre

**Comment by MathiasReker** [comment]
`@FIXME` = `@todo` ? :trollface:

File: src/PrestaShopBundle/Install/Upgrade.php [hunk 1]
```diff
@@ -250,10 +250,12 @@ private function defineConst()
             if (empty($this->oldVersion)) {
                 $this->oldVersion = Configuration::get('PS_INSTALL_VERSION');
             }
-            // fix : complete version number if there is not all 4 numbers
-            // for example replace 1.4.3 by 1.4.3.0
-            // consequences : file 1.4.3.0.sql will be skipped if oldversion = 1.4.3
-            // @since 1.4.4.0
+            /**
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Install/Upgrade.php]: Phpcs say you break something here :)

Comment (MathiasReker) [reply src/PrestaShopBundle/Install/Upgrade.php]: Can the order be the issue, since syntax is `@since [version] [<description>]` ?

**Comment by MathiasReker** [comment]
Still a issue with Travis, can you help me? :)

File: classes/Page.php [hunk 2]
```diff
@@ -54,8 +54,11 @@ public static function getCurrentId()
         $controller = Dispatcher::getInstance()->getController();
         $pageTypeId = Page::getPageTypeByName($controller);
 
-        // Some pages must be distinguished in order to record exactly what is being seen
-        // @todo dispatcher module
+        /**
+         * Some pages must be distinguished in order to record exactly what is being seen
```
Comment (jolelievre) [inline classes/Page.php:58]: missing ending .

File: classes/Product.php [hunk 3]
```diff
@@ -4967,12 +4967,13 @@ public static function getProductsProperties($id_lang, $query_result)
...
     }
 
-    /*
-    * Select all features for a given language
-    *
-    * @param $id_lang Language id
-    * @return array Array with feature's data
-    */
+    /**
+     * Select all features for a given language
```
Comment (jolelievre) [inline classes/Product.php:4971]: missing ending .

File: classes/Tools.php [hunk 4]
```diff
@@ -756,8 +756,9 @@ public static function displayPrice($price, $currency = null, $no_utf8 = false,
         return $cldr->getPrice($price, is_array($currency) ? $currency['iso_code'] : $currency->iso_code);
     }
 
-    /*
+    /**
      * Return a number well formatted
+     *
```
Comment (jolelievre) [inline classes/Tools.php:761]: missing ending .

File: classes/module/Module.php [hunk 5]
```diff
@@ -1967,7 +1967,7 @@ public function l($string, $specific = false, $locale = null)
         return Translate::getModuleTranslation($this, $string, ($specific) ? $specific : $this->name);
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/module/Module.php:1970]: missing ending .

File: classes/module/Module.php [hunk 6]
```diff
@@ -2027,7 +2027,7 @@ public function updatePosition($id_hook, $way, $position = null)
         return true;
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/module/Module.php:2030]: missing ending .

File: classes/module/Module.php [hunk 7]
```diff
@@ -2170,11 +2170,12 @@ public function displayInformation($information)
         return $output;
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/module/Module.php:2173]: missing ending .

File: classes/module/Module.php [hunk 8]
```diff
@@ -2227,10 +2228,11 @@ public static function getExceptionsStatic($id_module, $id_hook, $dispatch = fal
         return $array_return;
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/module/Module.php:2231]: missing ending .

File: classes/module/Module.php [hunk 9]
```diff
@@ -2806,10 +2808,11 @@ public function getPathUri()
         return $this->_path;
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/module/Module.php:2811]: missing ending .

File: classes/order/OrderDetail.php [hunk 10]
```diff
@@ -461,8 +461,9 @@ public static function getTaxListStatic($id_order_detail)
         return Db::getInstance()->executeS($sql);
     }
 
-    /*
+    /**
```
Comment (jolelievre) [inline classes/order/OrderDetail.php:464]: missing ending .

File: controllers/admin/AdminImportController.php [hunk 11]
```diff
@@ -1020,12 +1020,13 @@ protected function getTypeValuesOptions($nb_c)
         return $options;
     }
 
-    /*
-    * Return fields to be display AS piece of advise
-    *
-    * @param $in_array boolean
-    * @return string or return array
-    */
+    /**
```
Comment (jolelievre) [inline controllers/admin/AdminImportController.php:1023]: missing ending .

File: src/Adapter/Module/Tab/ModuleTabRegister.php [hunk 12]
```diff
@@ -275,9 +275,12 @@ protected function registerTab(Module $module, ParameterBag $tabDetails)
     {
         $this->checkIsValid($module->get('name'), $tabDetails);
 
-        // Legacy Tab, to be replaced with Doctrine entity when right management
-        // won't be directly linked to the tab creation
-        // @ToDo
+        /**
```
Comment (jolelievre) [inline src/Adapter/Module/Tab/ModuleTabRegister.php:278]: missing ending .

File: src/PrestaShopBundle/Install/Upgrade.php [hunk 13]
```diff
@@ -250,10 +250,10 @@ private function defineConst()
             if (empty($this->oldVersion)) {
                 $this->oldVersion = Configuration::get('PS_INSTALL_VERSION');
             }
-            // fix : complete version number if there is not all 4 numbers
-            // for example replace 1.4.3 by 1.4.3.0
-            // consequences : file 1.4.3.0.sql will be skipped if oldversion = 1.4.3
-            // @since 1.4.4.0
+            // Since 1.4.4.0
```
Comment (jolelievre) [inline src/PrestaShopBundle/Install/Upgrade.php:253]: Why not use the `/** */` form here?

File: src/PrestaShopBundle/Install/Upgrade.php [hunk 14]
```diff
@@ -262,7 +262,7 @@ private function defineConst()
             }
 
             $this->oldVersion = implode('.', $arrayVersion);
-            // end of fix
+            // End of fix
```
Comment (jolelievre) [inline src/PrestaShopBundle/Install/Upgrade.php:265]: Same question

**Review by jolelievre — CHANGES_REQUESTED** [review]
Thanks again @MathiasReker 
There are a few missing . at the end of PHPDocs
I think you just need to run php-cs-fixer to fix them
And a comment about `src/PrestaShopBundle/Install/Upgrade.php` but this one seems to have caused you trouble

**Comment by MathiasReker** [comment]
@jolelievre thanks for your review. Sorry, I did not explain my PR at all. I aim to fix php docs starting with `//`  and `/* ` in this PR. Once this is merged, I have some really nice improvements for the php docs, your comments are already covered by a script I prepared :) But I need this to be merged first please. I is better to split the PR's up in two.

File: src/PrestaShopBundle/Install/Upgrade.php (hunk 13 above)
Comment (MathiasReker) [reply src/PrestaShopBundle/Install/Upgrade.php:253]: Travis failed. I did not manage to solve the issue.

**Comment by jolelievre** [comment]
Ok let's not block this one for now then!

**Review by jolelievre — APPROVED** [review]
Second approval from @PrestaShop/prestashop-core-developers

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks everyone!

**Comment by MathiasReker** [comment]
Thanks!

**Pull request merged**